tests/examples_arguments_syntax/line_chart_with_cumsum_faceted.py

```diff
@@ -12,7 +12,7 @@
 columns_sorted = ['Drought', 'Epidemic', 'Earthquake', 'Flood']
 
 alt.Chart(source).transform_filter(
-    {'and': [
+    {'and': [ # type: ignore[arg-type]
```

Comment:
### Commit Message
> - Need to widen the definition of `_PredicateType`
> - To support this, we'll need to model with `TypedDict`(s)

Reply:
Wasn't expecting this to have ended up being so complex to solve.

I've tried a few things so far, but all seem to have trade-offs of varying degrees.
Will follow up with possible solutions in this thread

Reply:
# 1. Replace example with wrapper class

## Example(s)
Here we could use either:

<details><summary>Unpacked <code>dict</code> due to keyword conflict</summary>
<p>

```py
alt.Chart(source).transform_filter(
    alt.LogicalAndPredicate(
        **{
            "and": [
                alt.FieldOneOfPredicate(field="Entity", oneOf=columns_sorted),
                alt.FieldRangePredicate(field="Year", range=[1900, 2000]),
            ]
        }
    )
)
```

</p>
</details> 

<details><summary><code>SchemaBase.from_dict</code></summary>
<p>

```py
alt.Chart(source).transform_filter(
    alt.LogicalAndPredicate.from_dict(
        {
            "and": [
                alt.FieldOneOfPredicate(field="Entity", oneOf=columns_sorted),
                alt.FieldRangePredicate(field="Year", range=[1900, 2000]),
            ]
        }
    )
)
```

</p>
</details> 

## Pros

<details><summary>Passes existing type checking (<code>core.PredicateComposition</code> in <code>api._TestPredicateType</code>)</summary>
<p>

https://github.com/vega/altair/blob/211f7c517d7c005af521b242e51e414a084b8350/altair/vegalite/v5/api.py#L518-L522

</p>
</details> 

## Cons
- Incredibly verbose
- [User Guide-transform-filter](https://altair-viz.github.io/user_guide/transform/filter.html#logical-operands) would also need to be updated
	- Currently advises something that would introduce a warning at type checking time

Reply:
# 2. Prioritize merging (#3505)
Interestingly, this specific example already has a documented version using that syntax in (https://github.com/vega/altair/issues/3239#issuecomment-2252613018)

## Example
```py
alt.Chart(source).transform_filter(
    alt.field.one_of("Entity", columns_sorted) & alt.field.range("Year", [1900, 2000])
)
```

## Pros
- Passes existing type checking
- Incredibly terse

## Cons
I don't think enough progress has been made towards resolving the blocking issues I mentioned in the PR description:
- #3505
	- #3337
	- #2918

My concern is still one of **discoverability**

Reply:
# 3. Replace example using `*predicates`

## Example
```py
alt.Chart(source).transform_filter(
    alt.FieldOneOfPredicate(field="Entity", oneOf=columns_sorted),
    alt.FieldRangePredicate(field="Year", range=[1900, 2000]),
)
```

## Cons
**Not yet valid**, produces:

> Argument of type "FieldRangePredicate" cannot be assigned to parameter "more_predicates" of type "_ComposablePredicateType" in function "transform_filter"
> > Type "FieldRangePredicate" is not assignable to type "_ComposablePredicateType"
> > > "FieldRangePredicate" is not assignable to "OperatorMixin"
> > > "FieldRangePredicate" is not assignable to "SelectionPredicateComposition" Pylance(reportArgumentType)

Would require porting over [`_wrap_composition`](https://github.com/dangotbanned/altair/blob/f981130be22250a09b53f33b60b2c833af5028df/altair/vegalite/v5/_api_rfc.py#L95-L96):

```py
def _wrap_composition(predicate: Predicate, /) -> SelectionPredicateComposition:
    return SelectionPredicateComposition(predicate.to_dict())
```

Which would be called on a branch here for `core.Predicate`

https://github.com/vega/altair/blob/211f7c517d7c005af521b242e51e414a084b8350/altair/vegalite/v5/api.py#L762-L774

And updating the definition `api._ComposablePredicateType` to include `core.Predicate`

https://github.com/vega/altair/blob/211f7c517d7c005af521b242e51e414a084b8350/altair/vegalite/v5/api.py#L533-L535

Reply:
# 4. Type using `Mapping[Literal[...], ...]`

## Example

**Note that this is defined recursively**:

```py
_TestPredicateType: TypeAlias = Union[
    str,
    _expr_core.Expression,
    core.PredicateComposition,
    "_PredicateComposition",
]

_NotPredicate: TypeAlias = Mapping[Literal["not"], _TestPredicateType]
_AndPredicate: TypeAlias = Mapping[Literal["and"], Sequence[_TestPredicateType]]
_OrPredicate: TypeAlias = Mapping[Literal["or"], Sequence[_TestPredicateType]]

_PredicateComposition: TypeAlias = "_NotPredicate | _AndPredicate | _OrPredicate"

```

## Cons
Doesn't work, raises multiple issues

<details><summary><code>Mapping[str, Any]</code> invariant</summary>
<p>

Here it complains that `Literal["not"]` is not the same as `str`

![image](https://github.com/user-attachments/assets/cb117556-6c3b-4360-84df-ae713d56bbe9)

Also breaks exisiting tests:

![image](https://github.com/user-attachments/assets/9feb60be-f7fe-4da8-9586-e66a274ab357)


</p>
</details> 

<details><summary>Renders the annotation basically unusable</summary>
<p>

![image](https://github.com/user-attachments/assets/f04bce84-c743-40ed-b8f2-ac3f326deec7)

</p>
</details>

Reply:
Resolved in (https://github.com/vega/altair/pull/3664/commits/7f6c188a299560a03eddc594294e9869ae79f0f9)

Made possible by #3668